src/box/lua/config/configdata.lua

```diff
@@ -197,10 +198,54 @@ function methods.bootstrap_leader_name(self)
     return self._bootstrap_leader_name
 end
 
+-- Determine whether the instance will be started from an existing
+-- snapshot.
+function methods.has_snapshot(self)
+    return self._snapshot_path ~= nil
+end
```

Comment:
Nit: As for me, all these snapshot related functions are barely coupled with an existing configdata logic (which, mostly, just some views of a given config plus some config validation). I would add a file with snapshot search and analyzing into a new internal module in `src/box/lua/config/utils`.

Reply:
Great idea! Fixed